presto-main/src/main/java/com/facebook/presto/cost/ScalarStatsCalculator.java

```diff
@@ -106,5 +110,72 @@ protected SymbolStatsEstimate visitLiteral(Literal node, Void context)
             }
             return estimate.build();
         }
+
+        @Override
+        protected SymbolStatsEstimate visitArithmeticBinary(ArithmeticBinaryExpression node, Void context)
+        {
+            requireNonNull(node, "node is null");
+            SymbolStatsEstimate left = process(node.getLeft());
+            SymbolStatsEstimate right = process(node.getRight());
+
+            SymbolStatsEstimate.Builder result = SymbolStatsEstimate.builder()
+                    .setDataSize(Math.max(left.getDataSize(), right.getDataSize()))
+                    .setNullsFraction(left.getNullsFraction() + right.getNullsFraction() - left.getNullsFraction() * right.getNullsFraction())
+                    .setDistinctValuesCount(min(left.getDistinctValuesCount() * right.getDistinctValuesCount(), input.getOutputRowCount()));
```

Comment:
add `TODO` to drop the `min ... input.getOutputRowCount` when we have generic rule to do that after each computation.